subprojects/core/src/main/java/org/gradle/internal/cleanup/BuildOutputCleanupListener.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cleanup;
+
+import org.gradle.api.internal.GradleInternal;
+import org.gradle.initialization.ModelConfigurationListener;
+
+public class BuildOutputCleanupListener implements ModelConfigurationListener {
```

Comment:
could this happen any later? e.g., after the DAG has been created? I think we want to avoid doing work that will be counted as "configuration" time and it might also be useful to show something in the progress logger when we do this.